spf.go

```diff
@@ -1,20 +1,52 @@
 package spf
 
 import (
-	"fmt"
+	"errors"
 	"net"
 	"strconv"
 	"strings"
+)
 
-	"github.com/miekg/dns"
+// Errors could be used for root couse analysis
+var (
+	ErrDNSTemperror      = errors.New("temporary DNS error")
+	ErrDNSPermerror      = errors.New("permanent DNS error")
+	ErrInvalidDomain     = errors.New("invalid domain name")
+	errInvalidCIDRLength = errors.New("invalid CIDR length")
+	errDNSLimitExceeded  = errors.New("limit exceeded")
 )
```

Comment:
I am not a fan of such generic errors. As it's very easy to define them and later reuse in the code in my opinion is very information limited when it comes to debugging problems. You will know there is an invalid CIDR lenght, but impossible to know for what input. Now, was the error raise because the input is incorrect, there is a bug in the code or something else? I'd rather either leave old erroring mechanism or create corresponding functions, like ErrDNSTempError(s string) error { return fmt.Errorf("temporary DNS error for domain %s", s)} or something.
What do you think?